### ISSUE TITLE
Uncaught TypeError: cannot assign to readonly property '0' of string '1200'

### ISSUE THREAD

**Issue by akeating** [phoenixframework/phoenix_live_view#274]
### Environment

* Elixir version (elixir -v): 1.9.0
* Phoenix version (mix deps): 1.4.9
* phoenix_live_view at `957b7eb`

### Actual behavior
Page fails to render in the browser with console error:
`Uncaught TypeError: cannot assign to readonly property '0' of string '1200'`
<img width="878" alt="Screen Shot 2019-07-21 at 10 49 59 AM" src="https://user-images.githubusercontent.com/1427265/61594966-36927700-aba6-11e9-870f-90fe48a05ac2.png">

### Expected behavior
Page renders in the browser without console error.

### Analysis
These liveview templates fail on the *second* transition from `game_status: nil` to `game_status: :waiting_for_players`:

```
    def render(assigns) do
      ~L"""
      <%= render_game_panel(assigns) %>
      """
    end

    defp render_game_panel(%{ game_status: nil } = assigns) do
      ~L"""
      <div class="parking-panel">
        <div class="operations">
          <label>Game</label>
          <button phx-click="join-game-clicked">Play</button>
        </div>
      </div>
      """
    end      

    defp render_game_panel(%{ game_status: :waiting_for_players } = assigns) do
      ~L"""
      <div class="lobby-panel">
        <div class="operations">
          <label>Game</label>
          <div>Waiting for players...</div>
          <%= if assigns.player_count != nil do %>
            <div><%= assigns.player_count %> of <%= assigns.max_players_per_game %></div>
          <% end %>
          <div>Game will start when 8 players have joined</div>
          <button phx-click="start-game-clicked" <%= if !assigns.can_start, do: "disabled" %>>Request Start</button>
        </div>
      </div>
      """
    end
```

I identified two workarounds that might help with debugging:
1. Collapse the `render_game_panel` into top-level `render` functions
```
def render(%{ game_status: nil } = assigns) do
def render(%{ game_status: :waiting_for_players } = assigns) do
```
2. Remove the `<%= if assigns.player_count != nil do %>` conditional so the second render_game_panel function looks like this
```
defp render_game_panel(%{ game_status: :waiting_for_players } = assigns) do
  ~L"""
  <div class="lobby-panel">
    <div class="operations">
      <label>Game</label>
      <div>Waiting for players...</div>
      <div><%= assigns.player_count %> of <%= assigns.max_players_per_game %></div>
      <div>Game will start when 8 players have joined</div>
      <button phx-click="start-game-clicked" <%= if !assigns.can_start, do: "disabled" %>>Request Start</button>
    </div>
  </div>
  """
end
```
I am unblocked since I found workarounds but thought to let you know the details in the interests of hardening the codebase. Please let me know if I can help with any more information.

**Comment by chrismccord** [comment]
Can you provide a minimal complete LV module that reproduces the issue? It would really help us pinpoint where this is happening. Thanks!

**Comment by jayjun** [comment]
I encountered a very similar exception when updating a normal child to a LiveView one.

```
Uncaught TypeError: Cannot create property 'static' on string 'Normal'
```

Minimal code to reproduce here.

```elixir
defmodule MyAppWeb.BugLive do
  use Phoenix.LiveView

  def mount(_session, socket) do
    {:ok, assign(socket, live?: true)}
  end

  def render(assigns) do
    ~L"""
    <button phx-click="live?">Toggle</button>
    <%= render_child(assigns) %>
    """
  end

  def render_child(%{live?: true} = assigns) do
    ~L"Live"
  end

  def render_child(%{live?: false}) do
    "Normal"
  end

  def handle_event("live?", _value, socket) do
    {:noreply, assign(socket, live?: !socket.assigns.live?)}
  end
end
```

**Comment by jayjun** [comment]
Trust my luck to find another case, this time without child views but with `if`.

```
Uncaught TypeError: Cannot create property 'static' on string 'case'
```

To reproduce, nest an `if` in the first clause and a `case` in the second one.

```elixir
defmodule MyAppWeb.BugLive2 do
  use Phoenix.LiveView

  def mount(_session, socket) do
    {:ok, assign(socket, show_if: true)}
  end

  def render(assigns) do
    ~L"""
    <button phx-click="toggle">Toggle</button>

    <%= if @show_if do %>

      <%= if true do %>
        if
      <% end %>

    <% else %>

      <%= case true do %>
        <% true -> %>case
        <% _ -> %>case
      <% end %>

    <% end %>
    """
  end

  def handle_event("toggle", _value, socket) do
    {:noreply, assign(socket, show_if: !socket.assigns.show_if)}
  end
end
```

Both of my examples crash at the `recursiveMerge` function [here](https://github.com/phoenixframework/phoenix_live_view/blob/b44da5e96ffbb758584aed40348090779f9284ac/assets/js/phoenix_live_view.js#L240).

**Comment by snewcomer** [comment]
@jayjun Thanks for the example above!  Would it be possible to provide a repo so we can pull down and debug?  Want to make sure I'm seeing exactly what you are seeing.

**Comment by jayjun** [comment]
@snewcomer It appears 57e9e8d72995f7675510000394f15e16a2bcdce7 fixed the bug, or at least sidesteps it.

I’ve pushed https://github.com/jayjun/bug with LiveView pinned to a64b35509246914f0f86c47af70274bd727734d1 which demonstrates the problem so we can get to the bottom of this.

1. Start server and navigate to `http://localhost:4000`.
2. Click _Toggle_ to update LiveView.
3. Click _Toggle_ again and `Uncaught TypeError` will be thrown.

**Comment by jayjun** [comment]
@snewcomer It appears 57e9e8d fixed the bug, or at least sidesteps it.

I’ve pushed https://github.com/jayjun/bug with LiveView pinned to a64b355 which demonstrates the problem so we can get to the bottom of this.

1. Start server and navigate to http://localhost:4000.
2. Click _Toggle_ to update LiveView.
3. Click _Toggle_ again and LiveView won’t update. Check console for `TypeError` exceptions.

**Comment by chrismccord** [comment]
Fixed on master. Thanks!

**Comment by jayjun** [comment]
Fixed by #236.
